op-chain-ops/cmd/op-simulate/main.go

```diff
@@ -297,7 +296,10 @@ func simulate(ctx context.Context, logger log.Logger, conf *params.ChainConfig,
 
 	// run the transaction
 	start := time.Now()
-	receipt, err := core.ApplyTransaction(conf, cCtx, &sender, &gp, state, header, tx, &usedGas, vmConfig)
+	// TODO: is a nil author ok?
+	blockCtx := core.NewEVMBlockContext(header, cCtx, nil, conf, state)
+	evm := vm.NewEVM(blockCtx, state, conf, vmConfig)
+	receipt, err := core.ApplyTransaction(evm, &gp, state, header, tx, &usedGas)
```

Comment:
Wondering if the `txContext` has to be set here?

Reply:
No. The EVM tx context is set here: `ApplyTransaction` calls `ApplyTransactionExtended` calls `ApplyTransactionWithEVM` calls `ApplyMessage` calls `evm.SetTxContext`

And the state tx context (tx-hash, for logs collection) is already being set